source/appModules/foobar2000.py

```diff
@@ -14,31 +14,38 @@
 # A named tuple for holding the elapsed and total playing times from Foobar2000's status bar
 statusBarTimes = collections.namedtuple('StatusBarTimes', ['elapsed', 'total'])
 
-def getParsingFormat(interval):
-	"""Attempts to find a suitable parsing format string for a HH:MM:SS, MM:SS or SS -style time interval."""
-	timeParts = len(interval.split(":"))
+
+def parseIntervalToTimestamp(interval):
+	"""Parses a D HH:MM:SS, HH:MM:SS, MM:SS or SS -style interval to a timestamp."""
+	timeParts = len(interval.strip().replace(" ", ":").split(":"))
+	# day is automatically set to 1
+	# if no day is provided, remove 86400 seconds from parsed time
 	if timeParts == 1:
-		return "%S"
+		return calendar.timegm(time.strptime(interval.strip(), "%S")) - 86400
 	elif timeParts == 2:
-		return "%M:%S"
+		return calendar.timegm(time.strptime(interval.strip(), "%M:%S")) - 86400
 	elif timeParts == 3:
-		return "%H:%M:%S"
+		return calendar.timegm(time.strptime(interval.strip(), "%H:%M:%S")) - 86400
+	elif timeParts == 4:
+		return calendar.timegm(time.strptime(interval.strip(), "%dd %H:%M:%S"))
 	else:
 		return None
 
-def getOutputFormat(seconds):
-	"""Returns a format string for the given number of seconds with the least leading zeros."""
-	if seconds < 60:
-		return "%S"
-	elif seconds < 3600:
-		return "%M:%S"
+
+def getRemainingTime(parsedElapsedTime, parsedTotalTime):
+	"""Returns the difference between two times."""
+	remainingTime = parsedTotalTime - parsedElapsedTime
+	if remainingTime < 60:
+		return time.strftime("%S", time.gmtime(remainingTime))
+	elif remainingTime < 3600:
+		return time.strftime("%M:%S", time.gmtime(remainingTime))
+	elif remainingTime < 86400:
+		return time.strftime("%H:%M:%S", time.gmtime(remainingTime))
+	elif remainingTime < 172800:
+		return time.strftime("%#d day %H:%M:%S", time.gmtime(remainingTime - 86400))
 	else:
-		return "%H:%M:%S"
+		return time.strftime("%#d days %H:%M:%S", time.gmtime(remainingTime - 86400))

```

Comment:
"day"/"days" should also be translatable. the time format in general would be good to be translatable in general if possible.
I wonder if we have existing code in NVDA to make a localised timestamp announcement